File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int left = 0;
        for (int right = 1; right < nums.length; right++) {
            if (nums[left] != nums[right]) {
                left++;
                nums[left] = nums[right];//no changee will happen to the current right value . current value of right is moved to left.
            }
        }
        return left + 1;
    }
}

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
              
        left =0
        length = len(nums)
        for right in range(1,length):
            if (nums[left] != nums[right]):
                left =left +1
                nums[left]=nums[right]
            
        return left+1
                